Code/GraphMol/Depictor/EmbeddedFrag.cpp

```diff
@@ -1198,15 +1197,15 @@ void EmbeddedFrag::canonicalizeOrientation() {
 
   eig2.x = 2 * xy;
   eig2.y = (yy - xx) - d;
-  if (eig2.length() <= 1e-4) return;
   double eVal2 = (xx + yy - d) / 2;
-  eig2.normalize();
 
-  // make sure eig1 corresponds to the larger eigenvalue:
-  if (eVal2 > eVal1) {
-    RDGeom::Point2D tmp = eig1;
-    eig1 = eig2;
-    eig2 = tmp;
+  if (eig2.length() > 1e-4) {
+    eig2.normalize();
+
+    // make sure eig1 corresponds to the larger eigenvalue:
+    if (eVal2 > eVal1) {
+      std::swap(eig1, eig2);
+    }
   }
   // now rotate eig1 onto the X axis:
   trans.setVal(0, 0, eig1.x);
```

Comment:
It looks like the change is always doing the rotation?  I.e. not returning for small eig2's

Reply:
Yep, the bug was that rotation was not happening when eig2 was small